### ISSUE TITLE
KeyError: 'train_wall'

### ISSUE THREAD

**Issue by aj7tesh** [pytorch/fairseq#1678]
Everytime I reach epoch 2, at some interval I receive this error
2020-02-06 08:18:11 | INFO | train | epoch 002:   6149 / 11328 loss=427.878, nll_loss=425.296, ppl=1.06371e+128, wps=16218.2, ups=3.57, wpb=4546.2, bsz=172.2, num_updates=7842, lr=0.245094, gnorm=335.552, clip=100, oom=0, train_wall=1673, wall=0
2020-02-06 08:18:15 | INFO | fairseq.trainer | NOTE: overflow detected, (34, 'Numerical result out of range')
Traceback (most recent call last):
  File "train.py", line 11, in <module>
    cli_main()
  File "/home/ubuntu/fairseq-anuvaad/fairseq/fairseq_cli/train.py", line 307, in cli_main
    main(args)
  File "/home/ubuntu/fairseq-anuvaad/fairseq/fairseq_cli/train.py", line 102, in main
    train(args, trainer, task, epoch_itr)
  File "/home/ubuntu/anaconda3/lib/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/home/ubuntu/fairseq-anuvaad/fairseq/fairseq_cli/train.py", line 171, in train
    log_output = trainer.train_step(samples)
  File "/home/ubuntu/anaconda3/lib/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/home/ubuntu/fairseq-anuvaad/fairseq/fairseq/trainer.py", line 417, in train_step
    metrics.log_stop_time("train_wall")
  File "/home/ubuntu/fairseq-anuvaad/fairseq/fairseq/metrics.py", line 184, in log_stop_time
    agg[key].stop(weight)
KeyError: 'train_wall'

 - fairseq Version (e.g., 1.0 or master):
latest
 - PyTorch Version (e.g., 1.4)
 - OS (e.g., Linux):
 - Python version:
3.6.6
 - CUDA/cuDNN version:
10.2
 - GPU models and configuration:
GPU

**Comment by myleott** [comment]
I'm not able to reproduce this. "train_wall" is just the logged time spent in `Trainer.train_step`. We start the timer here: https://github.com/pytorch/fairseq/blob/d1c7eae1f5e0ed739dfe1acaa20ee675f613d6f5/fairseq/trainer.py#L280-L281 and stop it here: https://github.com/pytorch/fairseq/blob/master/fairseq/trainer.py#L417

That error seems to indicate we tried to stop it without starting it. Did you modify `train.py` or are you using the default fairseq one?

**Comment by risingdhxs** [comment]
I got the same problem while training a variant of the Transformer Big model on WMTbig dataset. It's hard to reproduce, since I'm running multiple models and yet only one got this error. It is not because the train_wall timer was not started, since I tried debugging and used printouts to ensure that the first log_start_time line was hit, even just before the error occurs. Commenting out these two lines seem to serve as a workaround, and after finishing the current checkpoint, remaining training can be resumed on the original code.

Felt like a hardware bug...
